src/fsharp/TastPickle.fs

```diff
@@ -1228,7 +1228,12 @@ let p_ident (x: Ident) st = p_tup2 p_string p_range (x.idText,x.idRange) st
 let p_xmldoc (XmlDoc x) st = p_array p_string x st
 
 let u_pos st = let a = u_int st in let b = u_int st in mkPos a b
-let u_range st = let a = u_string st in let b = u_pos st in let c = u_pos st in mkRange a b c
+
+let u_range st = 
+    let a = u_string st 
+    let b = u_pos st 
+    let c = u_pos st 
+    mkRange (if isNull a then null else try System.IO.Path.GetFullPath a with _ -> a) b c
```

Comment:
@dsyme Why I do this: file path which is got from assembly is the same as it was in the original project, which means it may be relative. So it goes to `range` and then such ranges are not equal when compared with ones gotten from projects which references it and "in memory cross projects reference" option is turned off. The comparison is performed like this (for methods, for example) https://github.com/Microsoft/visualfsharp/blob/master/src/fsharp/NameResolution.fs#L1324 (see the screenshot as well). If you see a better way, please point me at it.

Reply:
When can ``a`` be null, I've forgotten...?

But my main concern is that this is normalizing relative paths with respect to CurrentDirectory.  That's surely not what we want?  But equally I don't know what we do want - there is no "right" directory to use, is there?

I would be OK if the code explicitly only normalized non-relative paths, e.g. ones containing ``..\..`` as shown in the screenshot.  In fact, it feels like we should be doing that normalization when writing the ranges in the first place, or in the ``mkRange`` code

Reply:
@dsyme I moved the normalization to `mkRange` and perform it only on rooted paths.

Reply:
About nulls, I've removed the check because `Path.IsPathRooted` returns `false` for `null` path and everything works ok.